### PR TITLE
Protect routes and update tests.

### PR DESCRIPTION
**Pull request by neonleann**
https://trello.com/c/Z4QQdG76/1679-infosec-restrain-our-route-parameters-to-a-type-usually-ints-vv-e

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -44,22 +44,22 @@
             'as' => 'api.trader',
             'uses' => 'TraderController@show',
             // $user and App\Trader sent implicitly to policy.
-        ]);
+        ]) ->where('trader', '^[0-9]+$');
```
Comment (charlesstrange2) [inline routes/api.php:47]: Remove space between ")" and "->"

**Review by charlesstrange2 — CHANGES_REQUESTED** [review]
Got some spaces in places 👾

File: routes/api.php (hunk 1 above)
Comment (neonleann) [reply routes/api.php]: Silly me. All removed.

**Review by charlesstrange2 — APPROVED** [review]
okeodoke, give it a go

**Pull request merged**